csrc/ir/utils.cpp

```diff
@@ -1310,6 +1311,40 @@ MmaOpDetails getMmaOpDetails(
   return details;
 }
 
+namespace {
+std::vector<int> getMmaDimsPositions(MmaOp* mma) {
+  auto mma_domains = mma_utils::getProblemIterDomains(mma->fusion());
+  NVF_ERROR(mma_domains.isValid(), mma_domains.getErrorMsg());
+
+  const auto domains_data = mma_domains.getData();
+  const auto m_id = domains_data[(size_t)MatmulDomain::M];
+  const auto n_id = domains_data[(size_t)MatmulDomain::N];
+  const auto k_id = domains_data[(size_t)MatmulDomain::K];
+
+  std::optional<int> m_pos, n_pos, k_pos = std::nullopt;
```

Comment:
Nit: I'd simply use -1 to represent not-found. It's pretty clear that -1 isn't a valid index. 

```suggestion
  int m_pos = -1;
  int n_pos = -1;
  int k_pos = -1;
```